test/test_torch.py

```diff
@@ -6021,6 +6021,73 @@ def test_pow(self, device):
             torch.pow(m1, 1, out=out)
             self.assertEqual(out, m1)
 
+    @dtypes(*list(product(torch.testing.get_all_dtypes(include_bool=False),
+                          torch.testing.get_all_dtypes(include_bool=False))))
+    def test_float_power(self, device, dtypes):
+        base_dtype = dtypes[0]
+        exp_dtype = dtypes[1]
+        if base_dtype.is_complex or exp_dtype.is_complex:
+            dtype = torch.complex128
+        else:
+            dtype = torch.float64
+
+        base = torch.randint(-100, 100, (30,), device=device).to(dtype=base_dtype)
+        base[0] = base[8] = base[9] = 0
+        exp = torch.randint(-100, 100, (30,), device=device).to(dtype=exp_dtype)
+        exp[0] = exp[4] = exp[6] = 0
+
+        expected = torch.pow(base.to(dtype=dtype), exp.to(dtype=dtype))
+
+        result = torch.float_power(base, exp)
+        self.assertEqual(expected, result)
+
+        result = base.float_power(exp)
+        self.assertEqual(expected, result)
+
+        out = torch.empty_like(base).to(device=device, dtype=dtype)
+        torch.float_power(base, exp, out=out)
+        self.assertEqual(expected, out)
+
+        dtype_scalar_exp = torch.complex128 if base_dtype.is_complex else torch.double
```

Comment:
I'm not really sure what this section of the test is supposed to catch?

Reply:
The previous test was a bit confusing, I have divided the test for complex into a separate case.